engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/gateway/InclusiveGatewayTest.java

```diff
@@ -0,0 +1,1069 @@
+            tuple(PROCESS_ID, ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(PROCESS_ID, ProcessInstanceIntent.ELEMENT_COMPLETED));
+  }
+
+  @Test
+  public void shouldCompleteScopeOnInclusiveGateway() {
+    // given
+    final String processId = Strings.newRandomValidBpmnId();
+    final BpmnModelInstance process =
+        Bpmn.createExecutableProcess(processId)
+            .startEvent("start")
+            .sequenceFlowId("flow1")
+            .inclusiveGateway("inclusive")
+            .done();
+
+    ENGINE.deployment().withXmlResource(process).deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(processId).create();
+
+    // then
+    final List<Record<ProcessInstanceRecordValue>> processInstanceEvents =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .limitToProcessInstanceCompleted()
+            .asList();
+
+    assertThat(processInstanceEvents)
+        .extracting(e -> e.getValue().getElementId(), Record::getIntent)
+        .containsSequence(
+            tuple("inclusive", ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(processId, ProcessInstanceIntent.COMPLETE_ELEMENT));
+  }
+
+  @Test
+  public void shouldSplitInclusiveBranchesWithServiceTasks() {
+    // given
+    ENGINE.deployment().withXmlResource(INCLUSIVE_JOIN_PROCESS).deploy();
+
+    // when
+    final long processInstanceKey1 =
+        ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).withVariable("str", "a,b,c").create();
+    ENGINE.job().ofInstance(processInstanceKey1).withType("type1").complete();
+
+    // then
+    ENGINE.job().ofInstance(processInstanceKey1).withType("type2").complete();
+    ENGINE.job().ofInstance(processInstanceKey1).withType("type3").complete();
+    final List<Record<ProcessInstanceRecordValue>> events =
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey1)
+            .limitToProcessInstanceCompleted()
+            .asList();
+
+    assertThat(events)
+        .describedAs(
+            "after all sequence flows are taken, the joining inclusive gateway is activated exactly once")
+        .extracting(e -> e.getValue().getElementId(), Record::getIntent)
+        .containsSubsequence(
+            tuple("flow1", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING))
+        .containsSubsequence(
+            tuple("flow2", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING))
+        .containsSubsequence(
+            tuple("flow3", ProcessInstanceIntent.SEQUENCE_FLOW_TAKEN),
+            tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING))
+        .contains(tuple("join", ProcessInstanceIntent.ELEMENT_ACTIVATING));
+  }
+
+  @Test
+  public void shouldSplitInclusiveBranchesWithSomeServiceTask() {
```

Comment:
❌ This test contains a joining inclusive gateway (i.e. with multiple incoming sequence flows), which we won't support yet. Please remove this test.